packages/miniflare/src/plugins/hyperdrive/index.ts

```diff
@@ -115,15 +115,66 @@ export const HYPERDRIVE_PLUGIN: Plugin<typeof HyperdriveInputOptionsSchema> = {
 			})
 		);
 	},
-	async getServices({ options }) {
-		return Object.entries(options.hyperdrives ?? {}).map<Service>(
-			([name, url]) => ({
+	async getServices({ options, hyperdriveProxyController }) {
+		const services = [];
+		for (const [name, url] of Object.entries(options.hyperdrives ?? {})) {
+			const scheme = url.protocol.replace(":", "");
+
+			const proxyPort = await hyperdriveProxyController.createProxyServer({
+				name,
+				targetHost: url.hostname,
+				targetPort: getPort(url),
+				scheme,
+				sslmode: parseSslMode(url, scheme),
+			});
+			services.push({
 				name: `${HYPERDRIVE_PLUGIN_NAME}:${name}`,
 				external: {
-					address: `${url.hostname}:${getPort(url)}`,
+					address: `localhost:${proxyPort}`,
 					tcp: {},
 				},
-			})
-		);
+			});
+		}
+		return services;
 	},
 };
+
+// Postgres sslmode docs: https://www.postgresql.org/docs/current/libpq-ssl.html
+// MySQL ssl-mode docs: https://dev.mysql.com/doc/refman/8.4/en/using-encrypted-connections.html
+function parseSslMode(url: URL, scheme: string): string {
```

Comment:
Question: should search params always be normalized (to either casing)?

It's only done for some code paths.